### PR TITLE
dist/tools/coccinelle: annotate errors in Github Action

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This provides Github annotation support for the `coccinelle` check.

### Testing procedure
I provided a test commit that will cause some annotations.

### Issues/PRs references
None, but the check-script adaptation is similar to #15746 as both generate patches.

### CODE REVIEW

**Comment by aabadie** [comment]
I allowed myself to edit the PR as it was mentioning the headerguard check :)

**Comment by miri64** [comment]
> I allowed myself to edit the PR as it was mentioning the headerguard check :)

Ooops, copy-pasta...

File: sys/phydat/phydat.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ static const uint32_t divisors[] = {
 
 void phydat_fit(phydat_t *dat, const int32_t *values, unsigned int dim)
 {
-    assert(dim <= ARRAY_SIZE(dat->val));
+    assert(dim <= (sizeof(dat->val)/sizeof(dat->val[0])));
```
Comment (miri64) [inline sys/phydat/phydat.c:64]: Mhh curious that this isn't caught, but fixing this is not really part of this PR.

File: core/msg.c [hunk 2]
```diff
@@ -372,6 +373,7 @@ static int _msg_receive(msg_t *m, int block)
     list_node_t *next = list_remove_head(&me->msg_waiters);
 
     if (next == NULL) {
+        next++;
```
Comment (miri64) [inline core/msg.c:376]: This too

File: core/msg.c [hunk 3]
```diff
@@ -255,6 +255,7 @@ int msg_send_bus(msg_t *m, msg_bus_t *bus)
 
         if (_msg_send_oneway(m, subscriber->pid) > 0) {
             ++count;
+            continue;
```
Comment (miri64) [inline core/msg.c:258]: And this.

**Comment by aabadie** [comment]
Looks good to me (from the GH action point of view). Please drop the unrelated commits and squash.

**Comment by miri64** [comment]
> Looks good to me (from the GH action point of view). Please drop the unrelated commits and squash.

Let me just check first, if the problem I mentioned in #15748 exists here as well.

**Comment by miri64** [comment]
Problem [confirmed](https://github.com/RIOT-OS/RIOT/pull/15747/checks?check_run_id=1688344465#step:5:33). When the coccinelle check hits more than one time in the same file it only generates an annotation for the first hit with a patch to fix both. It still works, but I don't think it is nice. I will fix it like I did in #15748.

**Comment by miri64** [comment]
Now there are [two separate annotations](https://github.com/RIOT-OS/RIOT/runs/1688408628?check_suite_focus=true#step:5:33). Will squash now!

**Comment by miri64** [comment]
Done.

**Review by aabadie — APPROVED** [review]
ACK

**Comment by miri64** [comment]
Thanks for the review!
